lib/kube/token/validator.go

```diff
@@ -125,7 +147,7 @@ func (v *TokenReviewValidator) Validate(ctx context.Context, token, clusterName
 			// But people kept confusing it with JWKS and set the cluster name
 			// as the audience/. To avoid his common footgun we now allow tokens
 			// whose audience is the teleport cluster name.
-			Audiences: []string{kubernetesAudience, clusterName},
+			Audiences: append(audiences, clusterName),
```

Comment:
I don't want to think about potential excess capacity overwriting 😬 

```suggestion
			Audiences: append([]string{clusterName}, audiences...),
```

Reply:
Alternative solution: `v.clusterAudiences = slices.Clip(audiences)` at the point where we store the slice, so we don't have to think about it later.